File: Assignment/Abstraction/AssignmentNode.cs
namespace Assignment.Abstraction
{
    internal struct AssignmentNode : IExpressionNode
    {
        public AssignmentNode(INode left, INode right)
        {
            this.Left = left;
            this.Right = right;
        }
        public INode Left { get; }
        public INode Right { get; }

        public string DisplayName()
        {
            return $"{nameof(AssignmentNode)}";
        }
    }
}

File: src/main/java/ce305/abstraction/expressions/AssignmentNode.java
package ce305.abstraction.expressions;

import java.util.UUID;

import ce305.abstraction.AssignmentOperation;
import ce305.abstraction.INode;

public class AssignmentNode extends INode {
    public final INode left;
    public final INode right;
    public final AssignmentOperation operation;

    public AssignmentNode(INode left, INode right, AssignmentOperation operation) {
        super();
        this.left = left;
        this.right = right;
        this.operation = operation;
    }

    public AssignmentNode(INode left, INode right, AssignmentOperation operation, UUID id) {
        super(id);
        this.left = left;
        this.right = right;
        this.operation = operation;
    }

    @Override
    public String getDisplayName() {
        return String.format("%s(operation:%s)", AssignmentNode.class.getSimpleName(), this.operation.toString());
    }
}